src/m365/planner/commands/task/task-checklistitem-remove.spec.ts

```diff
@@ -0,0 +1,305 @@
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+
+    promptOptions = undefined;
+
+    sinon.stub(Cli, 'prompt').callsFake((options: any, cb: (result: { continue: boolean }) => void) => {
+      promptOptions = options;
+      cb({ continue: true });
+    });
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.patch,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      Cli.getInstance().config.all
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.TASK_CHECKLISTITEM_REMOVE), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['id', 'title', 'isChecked']);
+  });
+
+  it('prompts before removal when confirm option not passed', (done) => {
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake((options: any, cb: (result: { continue: boolean }) => void) => {
+      promptOptions = options;
+      cb({ continue: false });
+    });
+
+    command.action(logger, {
+      options: {
+        taskId: validTaskId,
+        id: validId
+      }
+    }, () => {
+      let promptIssued = false;
+
+      if (promptOptions && promptOptions.type === 'confirm') {
+        promptIssued = true;
+      }
+
+      try {
+        assert(promptIssued);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('passes validation when valid options specified', (done) => {
+    const actual = command.validate({
+      options: {
+        taskId: validTaskId,
+        id: validId
+      }
+    });
+    assert.strictEqual(actual, true);
+    done();
+  });
+
+  it('correctly removes checklistitem', (done) => {
+    sinon.stub(request, 'patch').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/planner/tasks/${encodeURIComponent(validTaskId)}/details`) {
+        return Promise.resolve(jsonOutput);
+      }
+
+      return Promise.reject('Invalid Request');
+    });
+
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/planner/tasks/${encodeURIComponent(validTaskId)}/details` &&
+        JSON.stringify(opts.headers) === JSON.stringify({
+          'accept': 'application/json'
+        })) {
+        return Promise.resolve({
+          "@odata.etag": "TestEtag"
+        });
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    const options: any = {
+      taskId: validTaskId,
+      id: validId,
+      confirm: true
+    };
+
+    command.action(logger, { options: options } as any, () => {
+      try {
+        assert(loggerLogSpy.calledWith(jsonOutput.checklist));
```

Comment:
When you remove the output from the command, this should also be updated to reflect that the logger is not called.

Reply:
Thank you. I have implemented the changes. Can you please review?